hamilton/plugins/h_diskcache.py

```diff
@@ -0,0 +1,126 @@
+    Remaining nodes may have multiple entries for different input values
+    """
+    nodes_history: Dict[str, List[str]] = cache.get(key=CacheHook.nodes_history_key)  # type: ignore
+
+    new_nodes_history = dict()
+    eviction_counter = 0
+    for node_name, history in nodes_history.items():
+        if len(history) < 1:
+            continue
+
+        if node_name in nodes_to_keep.keys():
+            node_to_keep = nodes_to_keep[node_name]
+            hash_to_keep = graph_utils.hash_source_code(node_to_keep.callable, strip=True)
+            history.remove(hash_to_keep)
+            new_nodes_history[node_name] = [hash_to_keep]
+
+        for hash_to_evict in history:
+            cache.evict(tag=f"{node_name}.{hash_to_evict}")
+            eviction_counter += 1
+
+    cache.set(key=CacheHook.nodes_history_key, value=new_nodes_history)
+    return eviction_counter
+
+
+def evict_all_except_driver(dr: driver.Driver) -> dict:
+    """Wrap the utility `evict_all_except` to receive a driver.Driver object"""
+    cache_hooks = [adapter for adapter in dr.adapter.adapters if isinstance(adapter, CacheHook)]
+
+    if len(cache_hooks) == 0:
+        raise AssertionError("0 `h_diskcache.CacheHook` defined for this Driver")
+    elif len(cache_hooks) > 1:
+        raise AssertionError(">1 `h_diskcache.CacheHook` defined for this Driver")
+
+    cache: diskcache.Cache = cache_hooks[0].cache
+    volume_before = cache.volume()
+    eviction_counter = evict_all_except(nodes_to_keep=dr.graph.nodes, cache=cache)
+    volume_after = cache.volume()
+    volume_difference = volume_before - volume_after
+
+    logger.info(f"Evicted: {_kb_to_mb(volume_difference):.2f} MB")
+    logger.debug(f"Evicted {eviction_counter} entries")
+    logger.debug(f"Cache size after: {_kb_to_mb(volume_after):.2f} MB")
+
+    return dict(
+        evicted_size_mb=_kb_to_mb(volume_difference),
+        eviction_counter=eviction_counter,
+        size_after=_kb_to_mb(volume_after),
+    )
+
+
+class CacheHook(
+    lifecycle.NodeExecutionHook,
+    lifecycle.GraphExecutionHook,
+    lifecycle.NodeExecutionMethod,
+):
+    nodes_history_key: str = "_nodes_history"
+
+    def __init__(
+        self, cache_vars: Union[List[str], None] = None, cache_path: str = ".", **cache_settings
+    ):
+        self.cache_vars = cache_vars if cache_vars else []
+        self.cache_path = cache_path
+        self.cache = diskcache.Cache(directory=cache_path, **cache_settings)
+        self.nodes_history: Dict[str, List[str]] = self.cache.get(
+            key=CacheHook.nodes_history_key, default=dict()
+        )  # type: ignore
+        self.used_nodes_hash: Dict[str, str] = dict()
+
+    def run_before_graph_execution(self, *, graph: graph_types.HamiltonGraph, **kwargs):
+        """Set cache_vars to all nodes if not specified"""
+        if self.cache_vars == []:
+            self.cache_vars = [n.name for n in graph.nodes]
+
+    def run_to_execute_node(
```

Comment:
Just worth noting that this can't work with other graph adapters that implement this -- should be fine though.